generator/src/main/java/com/scottlogic/deg/generator/generation/IsinStringGenerator.java

```diff
@@ -65,9 +115,15 @@ public boolean match(String subject) {
         }
         final List<Iterable<String>> countryCodeIterables = getAllCountryIsinGeneratorsAsStream()
             .limit(2)
-            .map(isinSansCheckDigitGenerator ->
-                new ProjectingIterable<>(isinSansCheckDigitGenerator.generateInterestingValues(),
-                    isinSansCheckDigit -> isinSansCheckDigit + IsinUtils.calculateIsinCheckDigit(isinSansCheckDigit)))
+            .map(isinGenerator ->
+                new FilteringIterable<>(
+                    new ProjectingIterable<>(
+                        isinGenerator.generateInterestingValues(),
```

Comment:
Repeated in the other class

Reply:
Which other class?

Reply:
`        return new FilteringIterable<>(
            new ProjectingIterable<>(
                regexGenerator.generateInterestingValues(),
                this::fixCheckDigit
            ),
            regexGenerator::match
        );`

or very similar excerpts are used here and in `ChecksummedCodeStringGenerator`